### PR TITLE
support pairwise h test #308; use testthat 3rd ed

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
- adds initial support for pairwise h-test objects
- tests for the same
- `NEWS.md` modified and adopts 80-characters limit
- adopts testthat 3.0 conventions
- retains only stricter tests for aov-ci functionality (since this tests all aspects of the output)

### CODE REVIEW

File: tests/testthat/test-betareg.R [hunk 1]
```diff
@@ -1,9 +1,13 @@
 if (Sys.getenv("USER") != "travis") {
   if (require("testthat") && require("parameters") && require("betareg")) {
+    library(betareg)
```
Comment (strengejacke) [inline tests/testthat/test-betareg.R:3]: we don't need the library() call here

File: tests/testthat/test-betareg.R [hunk 2]
```diff
@@ -37,12 +41,20 @@ if (Sys.getenv("USER") != "travis") {
     test_that("p_value", {
       expect_equal(
         p_value(m1)$p,
-        c(0, 0, 0, 0, 0, 0, 0, 0, 1e-05, 0.00114, 0, 6e-05),
+        c(
+          3.44252055464711e-250, 2.59222187511485e-65, 3.33610354513962e-29,
```
Comment (strengejacke) [inline tests/testthat/test-betareg.R:45]: why did you add this precision?

File: tests/testthat/test-model_parameters.aov_es_ci.R [hunk 3]
```diff
@@ -1,614 +1,436 @@
-if (require("insight") && require("effectsize") && require("testthat") && require("lme4") && require("parameters")) {
```
Comment (strengejacke) [inline tests/testthat/test-model_parameters.aov_es_ci.R:1]: why have these been removed?

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: tests/testthat/test-model_parameters.aov_es_ci.R (hunk 3 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-model_parameters.aov_es_ci.R:1]: I am not sure what these tests add anything over and above the stricter tests. The stricter tests covers all aspects of the returned output.

File: tests/testthat/test-betareg.R (hunk 2 above)
Comment (IndrajeetPatil) [reply tests/testthat/test-betareg.R:45]: Because the tests were failing with `testthat 3.0.0` and doing this fixed it (at least on my Windows machine).

Comment (strengejacke) [reply tests/testthat/test-betareg.R:45]: Ok, got it. But maybe you should not skip them everywhere? It should at least run on GitHub-CI.

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: tests/testthat/test-model_parameters.aov_es_ci.R (hunk 3 above)
Comment (strengejacke) [reply tests/testthat/test-model_parameters.aov_es_ci.R:1]: Ok, got it. But maybe you should not skip them everywhere? It should at least run on GitHub-CI.

**Comment by strengejacke** [comment]
And there's a generated PDF in the testthat folder that should be removed. Was it from local tests?

**Comment by IndrajeetPatil** [comment]
Why should it be removed? 
Running `devtools::test()` locally is always going to generate that file, so we might as well make peace with it. 
Or remove the tests that generate graphics.

**Comment by strengejacke** [comment]
Not sure if we need to add this file to rbuildignore? I personally don't like these temporary files ;-)

**Comment by IndrajeetPatil** [comment]
Yeah, me neither. But it's a pain to keep remembering to delete the file each time. Just ignoring it is better.

**Comment by IndrajeetPatil** [comment]
Is there a way to skip a certain test only on Ubuntu?

**Comment by strengejacke** [comment]
> Yeah, me neither. But it's a pain to keep remembering to delete the file each time. Just ignoring it is better.

Better to put them on gitignore, actually. Just came to my mind.

**Comment by strengejacke** [comment]
> Is there a way to skip a certain test only on Ubuntu?

```r
osx <- tryCatch({
  si <- Sys.info()
  if (!is.null(si["sysname"])) {
    si["sysname"] == "Darwin" || grepl("^darwin", R.version$os)
  } else {
    FALSE
  }
})
```

replace with Ubuntu-OS.

**Comment by strengejacke** [comment]
Maybe you should open a separate PR kust for the support of pairwise.htest, so that can be merged. Not sure if testthat 3rd ed. is stable enough for the CI?

**Comment by IndrajeetPatil** [comment]
But this failure actually has nothing to do with switching to `testthat 3.0`. 

This exact same thing was also happening [before](https://github.com/easystats/parameters/pull/324) and that is precisely why we were skipping these tests so far. 
So we must address this now.

**Comment by IndrajeetPatil** [comment]
Also, the test failure here: https://github.com/easystats/parameters/pull/327/checks?check_run_id=1370844389
has nothing to do with any of the changes I made. I am not sure what's going on there.

**Comment by IndrajeetPatil** [comment]
I think this can be merged, and we can conditionally skip the aov-ci related tests later. 
They don't fail either on `MacOS` or on `Windows`, so we can be confident that we are not breaking anything.

**Comment by IndrajeetPatil** [comment]
I saw you addressed this in a different PR. Should I just close this one then?

**Comment by strengejacke** [comment]
Yes, I copied your pairwise.htest-stuff to have this working, and we can open a new PR for the transition to testthat ed. 3. We should try to make all tests pass, as it is know.
